tfregistry/provider_test.go

```diff
@@ -0,0 +1,105 @@
+package tfregistry
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+func TestProviderLatest(t *testing.T) {
+	cases := []struct {
+		desc string
+		req  *ProviderLatestRequest
+		ok   bool
+		code int
+		res  string
+		want *ProviderLatestResponse
+	}{
+		{
+			desc: "simple",
+			req: &ProviderLatestRequest{
+				Namespace: "hashicorp",
+				Type:      "aws",
+			},
+			ok:   true,
+			code: 200,
+			res:  `{"version": "3.7.0", "versions": ["3.6.0", "3.5.0", "3.7.0"]}`,
+			want: &ProviderLatestResponse{
+				Version:  "3.7.0",
+				Versions: []string{"3.6.0", "3.5.0", "3.7.0"},
+			},
+		},
+		{
+			desc: "not found",
+			req: &ProviderLatestRequest{
+				Namespace: "hoge",
+				Type:      "piyo",
+			},
+			ok:   false,
+			code: 404,
+			res:  `{"errors":["Not Found"]}`,
+			want: nil,
+		},
+		{
+			desc: "invalid request (Namespace)",
+			req: &ProviderLatestRequest{
+				Namespace: "",
+				Type:      "piyo",
+			},
+			ok:   false,
+			code: 0,
+			res:  "",
+			want: nil,
+		},
+		{
+			desc: "invalid request (Name)",
```

Comment:
Why do we need this case? It seems to be valid dummy data (of course it's an invalid in the real world)

Reply:
I just did a poor copy/paste job from the module test case, removed.